router/router.go

```diff
@@ -0,0 +1,71 @@
+package router
+
+import (
+	"github.com/charmbracelet/bubbles/key"
+	tea "github.com/charmbracelet/bubbletea"
+)
+
+// Screen represents a model with a particular binding to trigger it
+type Screen struct {
+	model   tea.Model
+	binding key.Binding
```

Comment:
```suggestion
	binding key.Binding
	initialized bool
```

Reply:
Here we can keep track whether the screen has been initialized and every time we switch to a different screen check and `Init` the screen if it hasn't been.